dbms/src/Functions/FunctionsComparison.h

```diff
@@ -720,10 +734,41 @@ class FunctionComparison : public IFunction
         using ResultType = typename ResultColumnType::value_type;
         using StringImpl = StringComparisonWithCollatorImpl<Op<int, int>, ResultType>;
 
+        std::string_view c0_const_str_ref{};
+        std::string_view c1_const_str_ref{};
+
+        if (c0_const)
+        {
+            if (const auto * c0_const_string = checkAndGetColumn<ColumnString>(&c0_const->getDataColumn()); c0_const_string)
+            {
+                c0_const_str_ref = std::string_view(c0_const_string->getDataAt(0));
+            }
+            else if (const auto * c0_const_fixed_string = checkAndGetColumn<ColumnFixedString>(&c0_const->getDataColumn()); c0_const_fixed_string)
+            {
+                c0_const_str_ref = std::string_view(c0_const_fixed_string->getDataAt(0));
+            }
+            else
+                throw Exception("Logical error: ColumnConst contains not String nor FixedString column", ErrorCodes::ILLEGAL_COLUMN);
+        }
+
+        if (c1_const)
+        {
+            if (const auto * c1_const_string = checkAndGetColumn<ColumnString>(&c1_const->getDataColumn()); c1_const_string)
+            {
+                c1_const_str_ref = std::string_view(c1_const_string->getDataAt(0));
+            }
+            else if (const auto * c1_const_fixed_string = checkAndGetColumn<ColumnFixedString>(&c0_const->getDataColumn()); c1_const_fixed_string)
```

Comment:
should be c1_const->getDataColumn() here? Better to extract a method to do so.

Reply:
Fixed. That's a mistake. Thank you very much.